planning/obstacle_cruise_planner/include/obstacle_cruise_planner/node.hpp

```diff
@@ -138,14 +139,35 @@ class ObstacleCruisePlannerNode : public rclcpp::Node
 
   // subscriber
   rclcpp::Subscription<Trajectory>::SharedPtr traj_sub_;
-  rclcpp::Subscription<PredictedObjects>::SharedPtr objects_sub_;
-  rclcpp::Subscription<Odometry>::SharedPtr odom_sub_;
-  rclcpp::Subscription<AccelWithCovarianceStamped>::SharedPtr acc_sub_;
-
-  // data for callback functions
-  PredictedObjects::ConstSharedPtr objects_ptr_{nullptr};
-  Odometry::ConstSharedPtr ego_odom_ptr_{nullptr};
-  AccelWithCovarianceStamped::ConstSharedPtr ego_accel_ptr_{nullptr};
+  class PollingInputTopics
+  {
+  private:
+    tier4_autoware_utils::InterProcessPollingReceiver<Odometry> ego_odom_sub_;
+    tier4_autoware_utils::InterProcessPollingReceiver<PredictedObjects> objects_sub_;
+    tier4_autoware_utils::InterProcessPollingReceiver<AccelWithCovarianceStamped> acc_sub_;
+
+  public:
+    PollingInputTopics(rclcpp::Node * node)
+    : ego_odom_sub_(node, "~/input/odometry"),
+      objects_sub_(node, "~/input/objects"),
+      acc_sub_(node, "~/input/acceleration")
+    {
+    }
+    std::optional<Odometry> ego_odom_opt;
+    std::optional<PredictedObjects> objects_opt;
+    std::optional<AccelWithCovarianceStamped> ego_accel_opt;
+    bool takeData()
+    {
+      ego_odom_sub_.takeLastData(ego_odom_opt);
+      objects_sub_.takeLastData(objects_opt);
+      acc_sub_.takeLastData(ego_accel_opt);
+
+      if (!ego_odom_opt.has_value() || !objects_opt.has_value() || !ego_accel_opt.has_value()) {
+        return false;
+      }
+      return true;
+    };
+  } polling_topics_{this};
```

Comment:
The motivation to create a class in tier4_autoware_utils is not to create member variables for the data of each topic on the node side.
We can call `ego_odom_sub_.takeLastData` directly in the main callback of the node.

Reply:
@takayuki5168 
I think we should discuss aside

Reply:
delete this class and change to declare subscriber class on the node class